lib/ui/painting/gradient.cc

```diff
@@ -59,17 +59,26 @@ void CanvasGradient::initRadial(double center_x,
                                 double radius,
                                 const tonic::Int32List& colors,
                                 const tonic::Float32List& color_stops,
-                                SkShader::TileMode tile_mode) {
+                                SkShader::TileMode tile_mode,
+                                const tonic::Float64List& matrix4) {
   FXL_DCHECK(colors.num_elements() == color_stops.num_elements() ||
              color_stops.data() == nullptr);
 
   static_assert(sizeof(SkColor) == sizeof(int32_t),
                 "SkColor doesn't use int32_t.");
 
-  set_shader(SkGradientShader::MakeRadial(
-      SkPoint::Make(center_x, center_y), radius,
-      reinterpret_cast<const SkColor*>(colors.data()), color_stops.data(),
-      colors.num_elements(), tile_mode));
+  if (matrix4.data() == nullptr) {
+    set_shader(SkGradientShader::MakeRadial(
+        SkPoint::Make(center_x, center_y), radius,
+        reinterpret_cast<const SkColor*>(colors.data()), color_stops.data(),
+        colors.num_elements(), tile_mode));
+  } else {
+    SkMatrix sk_matrix = ToSkMatrix(matrix4);
+    set_shader(SkGradientShader::MakeRadial(
```

Comment:
Please dry up this call. You can pass in `nullptr` as the matrix argument if you don't get a valid matrix4 to convert.

Reply:
I think this removes the redundancy but adds some bloat. Any further recommendations?

```
  SkMatrix sk_matrix;
  bool has_matrix = matrix4.data() == nullptr;
  if (has_matrix) {
    sk_matrix = ToSkMatrix(matrix4);
  }

  set_shader(SkGradientShader::MakeRadial(
      SkPoint::Make(center_x, center_y), radius,
      reinterpret_cast<const SkColor*>(colors.data()), color_stops.data(),
      colors.num_elements(), tile_mode, 0, has_matrix ? &sk_matrix : nullptr));
```

Alternatively, since an SKMatrix is on the stack anyway, I could pass the identity when not provided, but that incurs an inversion and extra matrix multiplication internally in SKIA which would be nice to avoid (nullptr will be faster): https://github.com/google/skia/blob/0806b7ce3b594aef2b28bed8230a7923ed93415a/src/shaders/gradients/SkGradientShader.cpp#L828

Reply:
Don't have further ideas to make this elegant. This should be fine.